src/main/java/smile/identity/core/PartnerParameters.java

```diff
@@ -1,42 +1,57 @@
 package smile.identity.core;
+
 import org.json.simple.JSONObject;
 
 interface Parameters {
-  public void add(String key1, String key2);
-  public String get();
+    public void add(String key1, String key2) throws Exception;
```

Comment:
(key1, key2) should be (key, val). 
also, is there a need to inherit from this class?

Reply:
can these functions be moved to base class: checkNullAndEmpty(), checkNull()